exam/models/exam.py

```diff
@@ -31,30 +31,59 @@ class ExtendedTimeTable(models.Model):
         for rec in self:
-            rec.user_ids = [teacher.teacher_id.user_id.id
-                            for teacher in rec.timetable_ids
-                            if rec.timetable_type == 'exam']
-        return True
+            exam_search = exam.search([('state', '=', 'running')])
+            for data in exam_search:
+                schedule_line_search = schedule_line.search([('exam_id', '=',
+                                                              data.id),
+                                                             ('timetable_id',
+                                                              '=', rec.id)])
+                if schedule_line_search:
+                    raise ValidationError(_('''You cannot delete schedule
+                                           of exam in running state'''))
+        return super(ExtendedTimeTable, self).unlink()
 
     timetable_type = fields.Selection(selection_add=[('exam', 'Exam')],
                                       string='Time Table Type', required=True,
                                       inivisible=False)
     exam_timetable_line_ids = fields.One2many('time.table.line', 'table_id',
                                               'TimeTable')
     exam_id = fields.Many2one('exam.exam', 'Exam')
-    users = fields.Many2many('res.users', string="Users",
-                             compute="_compute_users", store=True)
+
+    @api.multi
+    @api.constrains('exam_timetable_line_ids')
+    def _check_exam(self):
+        '''Method to check same exam is not assigned on same day'''
+        if not self.exam_timetable_line_ids:
+            raise ValidationError(_('''Enter all details
+                                    for Exam Timetable'''))
+        if self.timetable_type == 'exam':
+            domain = [('table_id', 'in', self.ids)]
+            line_ids = self.env['time.table.line'].search(domain)
+            for rec in line_ids:
+                records = [rec_check.id for rec_check in line_ids
+                           if (rec.day_of_week == rec_check.day_of_week and
+                               rec.start_time == rec_check.start_time and
+                               rec.end_time == rec_check.end_time and
+                               rec.teacher_id.id == rec.teacher_id.id and
+                               rec.exm_date == rec.exm_date)]
+                if len(records) > 1:
+                    raise ValidationError(_('''You cannot set exam
+                                           at same time %s
+                                           at same day %s for teacher %s''') %
+                                          (rec.start_time, rec.day_of_week,
+                                           rec.teacher_id.name))
 
 
 class ExtendedTimeTableLine(models.Model):
     _inherit = 'time.table.line'
 
     exm_date = fields.Date('Exam Date')
     day_of_week = fields.Char('Week Day')
+    class_room_id = fields.Many2one('class.room', 'Classroom')
```

Comment:
Should not it be mandatory at view level atleast?

Reply:
done